frontend/packages/console-shared/src/components/actions/loader/ActionsLoader.tsx

```diff
@@ -2,80 +2,96 @@ import * as React from 'react';
 import * as _ from 'lodash';
 import {
   Action,
-  ActionGroup,
   ActionProvider,
-  isActionGroup,
   isActionProvider,
+  isResourceActionProvider,
+  ResourceActionProvider,
   useResolvedExtensions,
 } from '@console/dynamic-plugin-sdk';
-import { useExtensions } from '@console/plugin-sdk';
-import { MenuOption } from '../menu/menu-types';
-import { createMenuOptions } from '../menu/menu-utils';
+import { referenceForExtensionModel } from '@console/internal/module/k8s';
 import ActionsHookResolver from './ActionsHookResolver';
 
 type ActionsLoaderProps = {
   contextId: string;
   scope: any;
-  children: (loader: Loader) => React.ReactNode;
+  onActionsLoaded: (actions: Action[]) => void;
+  onContextChange: (contextId: string, scope: any) => void;
+  onLoadError: (error: any) => void;
 };
 
-type Loader = { actions: Action[]; options: MenuOption[]; loaded: boolean; error: any };
-
-const ActionsLoader: React.FC<ActionsLoaderProps> = ({ contextId, scope, children }) => {
+const ActionsLoader: React.FC<ActionsLoaderProps> = ({
+  contextId,
+  scope,
+  onActionsLoaded,
+  onContextChange,
+  onLoadError,
+}) => {
   const [actionsMap, setActionsMap] = React.useState<{ [uid: string]: Action[] }>({});
   const [loadError, setLoadError] = React.useState<any>();
 
   const onProviderValueResolved = React.useCallback((actions: Action[], uid: string) => {
     setActionsMap((prev) => ({ ...prev, [uid]: actions }));
   }, []);
 
-  const actionProviderGuard = React.useCallback(
+  const providerGuard = React.useCallback(
     (e): e is ActionProvider => {
       return isActionProvider(e) && e.properties.contextId === contextId;
     },
     [contextId],
   );
 
+  const resourceProviderGuard = React.useCallback(
+    (e): e is ResourceActionProvider => {
+      const modelKind = referenceForExtensionModel(e.properties.model);
+      return isResourceActionProvider(e) && modelKind === contextId;
```

Comment:
You must pass the guard check first of `isResourceActionProvider` before accessing `e.properties`.

`e` is currently of type `any` when it should be of type `Extension`

Reply:
Updated.